internal/serve/httphandler/mfa_handler.go

```diff
@@ -43,8 +44,36 @@ func (h MFAHandler) ServeHTTP(rw http.ResponseWriter, req *http.Request) {
 		return
 	}
 
+	deviceID := req.Header.Get(DeviceIDHeader)
+	if deviceID == "" {
+		httperror.BadRequest("Device-ID header is required", nil, nil).Render(rw)
+		return
+	}
+
+	if reqBody.MFACode == "" {
+		extras := map[string]interface{}{"mfa_code": "MFA Code is required"}
+		httperror.BadRequest("Request invalid", nil, extras).Render(rw)
+		return
+	}
+
+	user, err := h.AuthManager.GetUserByMFA(ctx, deviceID)
```

Comment:
👋  Why is this method called `GetUserByMFA` if what you're passing is the deviceID and not the MFA?